compiler/rustc_ast/src/ast.rs

```diff
@@ -2299,21 +2299,17 @@ impl FnRetTy {
     }
 }
 
-/// Module declaration.
-///
-/// E.g., `mod foo;` or `mod foo { .. }`.
+/// Module item kind.
 #[derive(Clone, Encodable, Decodable, Debug)]
-pub struct Mod {
-    /// A span from the first token past `{` to the last token until `}`.
-    /// For `mod foo;`, the inner span ranges from the first token
-    /// to the last token in the external file.
-    pub inner: Span,
-    /// `unsafe` keyword accepted syntactically for macro DSLs, but not
-    /// semantically by Rust.
-    pub unsafety: Unsafe,
-    pub items: Vec<P<Item>>,
-    /// `true` for `mod foo { .. }`; `false` for `mod foo;`.
-    pub inline: bool,
+pub enum ModKind {
+    /// Module with inlined definition `mod foo { ... }`.
+    /// The inner span is from the first token past `{` to the last token until `}`.
+    Inlined(Vec<P<Item>>, Span),
+    /// Module with definition outlined to a separate file `mod foo;` and already loaded from it.
+    /// The inner span ranges from the first to the last token in the loaded file.
+    OutlinedLoaded(Vec<P<Item>>, Span),
```

Comment:
Alternative 1: rename this to just `Outlined` for brevity, it's hard to mix up loaded and unloaded modules due to different fields.

Alternative 2: two-variant enum
```rust
enum ModKind {
    Loaded(Vec<P<Item>>, Span, IsInline),
    Unloaded,
}
```
This should be more convenient because `Inlined` and `OutlinedLoaded` are almost always treated in the same way (to obtain items or inner span from them, see the second commit). I'm not sure whether it would be more clear though.

Reply:
I have a slight preference for the two-variant enum, but I'll leave the final decision up to you.

Reply:
Switched to the alternative 2.